### ISSUE TITLE
flattening expected args doesn't work for methods that actually expect arrays

### ISSUE THREAD

**Issue by aaronjensen** [rentalcustard/matahari#10]
```

describe "bug" do
  it "should work" do
    foo = spy
    foo.test([2, 3])
    foo.should have_received.test([2, 3])
  end
end

```

**Comment by rentalcustard** [comment]
Ouch, good catch! Thanks for spotting this. Can you confirm it's fixed in v0.3.2?

**Comment by aaronjensen** [comment]
haven't had a chance to confirm arrays, but it seems that passing any sort of stub into a method breaks too: 

`can't convert RSpec::Mocks::Mock to Array (RSpec::Mocks::Mock#to_ary gives RSpec::Mocks::Mock)`

in line 7 of invocation.rb
